src/main/scala/freeeventsourcing/syntax/ProcessSyntax.scala

```diff
@@ -0,0 +1,218 @@
+      }
+    }
+
+    /** Helper class for from. */
+    final class FromAggregateBuilder[A <: Aggregate] private[ProcessSyntax] (aggregateType: A, aggregate: A#Id) {
+      /** Await an event from the aggregate. */
+      def await[E <: A#Event: AggregateEventType](implicit ev: ValidAggregate[BC, A], idser: StringSerializable[A#Id]) = {
+        val selector = AggregateEventSelector(aggregateType)(aggregate)[E]
+        ProcessSyntax.this.await(selector)
+      }
+    }
+
+    /** Helper class for execute. */
+    type CommandErrorHandler[Error <: Coproduct] = Error ⇒ ProcessMonad[Unit]
+    object CommandErrorHandler {
+      private[ProcessSyntax] def builder[E <: Coproduct]() = new Builder[E, E](e ⇒
+        throw new AssertionError("Error in CommandErrorHandler, type was constructed that does not " +
+          "handle all cases. Should have been prevented by the compiler"))
+
+      type EmptyBuilder[E <: Coproduct] = Builder[E, E]
+
+      final class Builder[All <: Coproduct, Unhandled <: Coproduct] private[CommandErrorHandler] (handled: CommandErrorHandler[All]) {
+        def catching[E](handler: E ⇒ ProcessMonad[Unit])(implicit ev: HandleError[Unhandled, E], s: Selector[All, E]) = {
+          new Builder[All, ev.Rest](error ⇒ s(error).map(handler).getOrElse(handled(error)))
+        }
+
+        /** Return the fully constructed error handler. Can only be called if all cases have been handled */
+        private[ProcessSyntax] def errorHandler(implicit ev: AllErrorsHandled[Unhandled]) = handled
+      }
+    }
+
+    /** Option inside a firstOf. */
+    sealed trait SwitchPath {
+      type Event
+      type Result
+      val action: Await[BC, Event]
+      def effect(event: Event): ProcessMonad[Result]
+    }
+    object SwitchPath {
+      type Aux[E, R] = SwitchPath { type Event = E; type Result = R }
+    }
+    /** All options of a firstOf. */
+    sealed trait Switch[Paths <: HList] {
+      type Events <: Coproduct
+      type A <: Alternatives[BC]
+      type Result <: Coproduct
+      def effectFor(paths: Paths)(event: A#Events): ProcessMonad[Result]
+      def alternatives(paths: Paths): A
+    }
+    object Switch {
+      type Aux[P <: HList, R <: Coproduct] = Switch[P] { type Result = R }
+
+      implicit def end = new Switch[HNil] {
+        type A = FirstOf.Empty[BC]
+        type Events = CNil
+        type Result = CNil
+        def effectFor(paths: HNil)(event: CNil) = throw new AssertionError("Cannot be reached, the compiler is supposed to prevent that")
+        def alternatives(paths: HNil) = FirstOf.Empty()
+      }
+      implicit def head[E, R, T <: HList](implicit t: Switch[T]) = new Switch[SwitchPath.Aux[E, R] :: T] {
+        type A = Alternative[BC, E, t.A]
+        type P = SwitchPath.Aux[E, R]
+        type Events = P#Event :+: t.Events
+        type Result = P#Result :+: t.Result
+        def effectFor(paths: P :: T)(event: A#Events) = event match {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method name does not match the regular expression '"^[a-z][A-Za-z0-9]*$"'](https://www.codacy.com/app/msiegenthaler/free-event-sourcing/file/2571398663/issues/source?bid=3298199&fileBranchId=3298202#l142)